File: app/src/main/java/oop1/ex02.java
package oop1;

public class ex02 {
    static class Student {
        String name;
        int age;

        Student() {
            this("Unknown", 20);
        }

        Student(String name, int age) {
            this.name = name;
            this.age = age;
        }

        @Override
        public String toString() {
            return "Student{name='" + name + "', age=" + age + "}";
        }
    }

    public static void main(String[] args) {
        System.out.println(new Student());
        System.out.println(new Student("Bob", 22));
    }
}

File: app-kotlin/src/main/kotlin/oop1/ex02.kt
package oop1

class ex02 {
    class Student(val name: String, val age: Int) {
        constructor() : this("Unknown", 20)
        override fun toString(): String = "Student{name='$name', age=$age}"
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            println(Student())
            println(Student("Bob", 22))
        }
    }
}
